### PR TITLE
Far point joints

### PR DESCRIPTION
**Pull request by sethalves**
- Add avatar pseudo-joints for far-grab targets
- create grab module for parenting-based far-grabs of non-dynamic entities

### CODE REVIEW

**Comment by sethalves** [comment]
to test:
- exercise all aspects of far-grabbing with observers

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11945/HighFidelity-Beta-PR13784-6076c40.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11945/HighFidelity-Beta-PR13784-6076c40.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11945/HighFidelity-Beta-Interface-PR13784-6076c40.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11945/HighFidelity-Beta-PR13784-6076c40.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11945/HighFidelity-Beta-Interface-PR13784-6076c40.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11978/HighFidelity-Beta-PR13784-707bbb1.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11980/HighFidelity-Beta-PR13784-5208e73.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11980/HighFidelity-Beta-PR13784-5208e73.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11980/HighFidelity-Beta-Interface-PR13784-5208e73.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11980/HighFidelity-Beta-PR13784-5208e73.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11980/HighFidelity-Beta-Interface-PR13784-5208e73.exe

File: interface/src/avatar/MyAvatar.cpp [hunk 1]
```diff
@@ -1578,6 +1643,8 @@ void MyAvatar::clearJointsData() {
         QMetaObject::invokeMethod(this, "clearJointsData");
         return;
     }
+    _farGrabRightMatrixCache.invalidate();
+    _farGrabLeftMatrixCache.invalidate();
```
[the remark this answers is not in the document]
Comment (sethalves) [reply interface/src/avatar/MyAvatar.cpp]: yes (done)

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11991/HighFidelity-Beta-PR13784-22894aa.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11994/HighFidelity-Beta-PR13784-4a92871.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12001/HighFidelity-Beta-PR13784-7fc7ed1.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12001/HighFidelity-Beta-PR13784-7fc7ed1.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12001/HighFidelity-Beta-Interface-PR13784-7fc7ed1.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12001/HighFidelity-Beta-PR13784-7fc7ed1.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12001/HighFidelity-Beta-Interface-PR13784-7fc7ed1.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12007/HighFidelity-Beta-PR13784-959f37f.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12007/HighFidelity-Beta-PR13784-959f37f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12007/HighFidelity-Beta-Interface-PR13784-959f37f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12007/HighFidelity-Beta-PR13784-959f37f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12007/HighFidelity-Beta-Interface-PR13784-959f37f.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qligmaloney** [comment]
Build this please

**Comment by qliacoombs** [comment]
Build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12096/HighFidelity-Beta-PR13784-527b27d.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12096/HighFidelity-Beta-PR13784-527b27d.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12096/HighFidelity-Beta-Interface-PR13784-527b27d.exe

**Comment by qliacoombs** [comment]
Build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12099/HighFidelity-Beta-PR13784-527b27d.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12099/HighFidelity-Beta-PR13784-527b27d.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12099/HighFidelity-Beta-Interface-PR13784-527b27d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12099/HighFidelity-Beta-PR13784-527b27d.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12099/HighFidelity-Beta-Interface-PR13784-527b27d.exe

**Comment by JeffClinton** [comment]
Currently testing

**Comment by JeffClinton** [comment]
PASS @Conklin94122

**Pull request merged**